### PR TITLE
Switching from Hibernate Mappings to Annotations - PersonAttributeType

### PR DESCRIPTION
**Pull request by mherman22**
## Description of what I changed

I changed the PersonAttributeType domain's reliance on xml mappings to use the more understandable JPA annotations. I used annotations like https://github.com/column, https://github.com/entity https://github.com/table @OneToMany and many others as can be seen in the changes in this pull request.

I tested the changes against a local setup using sdk by creating a person attribute type using this endpoint http://localhost:8091/openmrs/ws/rest/v1/personattributetype using the payload below;-
```
{
    "name": "Edit User Privileges",
    "description": "Able to manage the user privileges",
    "format": "org.openmrs.Concept",
    "foreignKey": 1054,
    "searchable": false,
    "editPrivilege": "24635eec-dd5a-11e6-9d9c-0242ac150002"
}
```
And the resulting payload is as shown below;-
```
{
    "uuid": "f8e16d41-a8b5-4a78-b696-7358bd90e7ac",
    "display": "Edit User Privileges",
    "name": "Edit User Privileges",
    "description": "Able to manage the user privileges",
    "format": "org.openmrs.Concept",
    "foreignKey": 1054,
    "sortWeight": 17.0,
    "searchable": false,
    "editPrivilege": null,
    "retired": false,
    "links": [{
        "rel": "self",
        "uri": "http://localhost:8091/openmrs/ws/rest/v1/personattributetype/f8e16d41-a8b5-4a78-b696-7358bd90e7ac",
        "resourceAlias": "personattributetype"
    }, {
        "rel": "full",
        "uri": "http://localhost:8091/openmrs/ws/rest/v1/personattributetype/f8e16d41-a8b5-4a78-b696-7358bd90e7ac?v=full",
        "resourceAlias": "personattributetype"
    }],
    "resourceVersion": "1.8"
}
```
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5945

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/PersonAttributeType.java [hunk 1]
```diff
@@ -9,31 +9,50 @@
...
  */
+@Entity
+@Table(name = "person_attribute_type")
 public class PersonAttributeType extends BaseChangeableOpenmrsMetadata implements java.io.Serializable, Comparable<PersonAttributeType> {
 	
 	public static final long serialVersionUID = 2112313431211L;
 	
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "person_attribute_type_id")
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/PersonAttributeType.java:38]: How are we capturing this? `unsaved-value="0"`

Comment (mherman22) [reply api/src/main/java/org/openmrs/PersonAttributeType.java]: So my thinking was that in the JPA annotation mapping, we don't need to explicitly set unsaved-value="0", as this behavior is already handled by the GenerationType.IDENTITY strategy. But let me add the `nullable=false`

File: api/src/main/java/org/openmrs/PersonAttributeType.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public class PersonAttributeType extends BaseChangeableOpenmrsMetadata implement
 	
 	@Id
 	@GeneratedValue(strategy = GenerationType.IDENTITY)
-	@Column(name = "person_attribute_type_id")
+	@Column(name = "person_attribute_type_id", nullable = false)
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/PersonAttributeType.java:38]: do we need nullable = false on an id? Isn't it automatically implied?

Comment (mherman22) [reply api/src/main/java/org/openmrs/PersonAttributeType.java]: i have gotten rid of it

**Pull request merged**